### PR TITLE
HHH-17662 Fix memory leak issue due to ArrayJdbcType not having equals/hashCode

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-17662

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java [hunk 1]
```diff
@@ -174,6 +178,62 @@ public AggregateJdbcType findAggregateDescriptor(String typeName) {
 		return aggregateDescriptorMap.get( typeName.toLowerCase( Locale.ROOT ) );
 	}
 
+	public JdbcType resolveTypeConstructorDescriptor(
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java:181]: Can we please add some Javadoc here.  Why the map of maps?

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java [hunk 2]
```diff
@@ -174,6 +178,62 @@ public AggregateJdbcType findAggregateDescriptor(String typeName) {
...
+			);
+			descriptors.put( elementType, jdbcType );
+			return jdbcType;
+		}
+		else {
+			return getDescriptor( jdbcTypeConstructorCode );
+		}
+	}
+
+	public JdbcType resolveTypeConstructorDescriptor(
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java:209]: Javadoc

**Review by sebersole — COMMENTED** [review]
Few comments.  Trying to grok the need for the map of maps here.

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java (hunk 1 above)
Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java:194]: You're right, it's better to have a single key object. I fixed that and added some java doc.

**Pull request merged**